tests/jobs/scala/src/main/scala/DatasetGenerator.scala

```diff
@@ -0,0 +1,57 @@
+import org.apache.spark.sql.SparkSession
+
+import scala.util.Random
+
+/**
+	* Generates dataset in form of randomly generated key/value pairs. Can be used in stress-tests and benchmarks
+  *
+	* Usage: DatasetGenerator [totalRecords] [recordSize] [s3_destination]
```

Comment:
I'd suggest adding the following parameters:
- `numFiles` to control the resulting number of files created (maps to the number of partitions)
- `staticPrefixLength` to control the length of the static part of the keys 
- `randomSiffuxLength` to control the length of the variable part of the keys
- rename `recordSize` -> `valueSizeBytes`